### PR TITLE
Use decimal precision for write-off

### PR DESCRIPTION
**Pull request by mdietrichc2c**
This PR is to use decimal precision to assess that the write-off is equal to 0; otherwise, you can have a write-off line with "-0.00" or "0.00" as an amount in the wizard.

Done in https://github.com/odoo/odoo/pull/4969 as well.

### CODE REVIEW

**Comment by vrenaville** [comment]
:+1:

**Comment by guewen** [comment]
IMO you should use `res_currency.is_zero()` so you use the precision of the currency of the lines, which may be different.

**Comment by mdietrichc2c** [comment]
@guewen correct; it now uses the line's currency precision (or the company currency, if none was defined).

**Comment by mdietrichc2c** [comment]
@guewen as discussed, I removed the account's currency (since the wizard doesn't compare amount_currency).

**Comment by guewen** [comment]
:+1:

**Comment by fclementic2c** [comment]
:+1:

File: addons/account/wizard/account_reconcile.py [hunk 1]
```diff
@@ -56,15 +58,22 @@ def trans_rec_get(self, cr, uid, ids, context=None):
...
             if not line.reconcile_id and not line.reconcile_id.id:
                 count += 1
                 credit += line.credit
                 debit += line.debit
-                account_id = line.account_id.id
-        return {'trans_nbr': count, 'account_id': account_id, 'credit': credit, 'debit': debit, 'writeoff': debit - credit}
+                account = line.account_id
+        # If the write-off is lower than the account precision,
+        # set it to 0.
+        currency = account.company_id.currency_id
```
Comment (gurneyalex) [inline addons/account/wizard/account_reconcile.py:74]: potentially account is False here and we will have an AttributeError exception. 

am I missing something?

**Comment by mdietrichc2c** [comment]
@gurneyalex no, you're correct, even if account will most probably never be False; I added a check to ensure this anyway.

File: addons/account/wizard/account_reconcile.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 from openerp.osv import fields, osv
 from openerp.tools.translate import _
 import openerp.addons.decimal_precision as dp
+from openerp.tools.float_utils import float_is_zero
```
Comment (hbrunn) [inline addons/account/wizard/account_reconcile.py]: this import seems to be unused

**Comment by mdietrichc2c** [comment]
@hbrunn correct, sorry; forgot to remove it when the method was changed to currency.is_zero

**Comment by hbrunn** [comment]
:+1:

**Pull request merged**